extensions/integration-tests/testfmks/src/stress.ts

```diff
@@ -0,0 +1,289 @@
+ * @param passThreshold - the fractional number of all invocations of the decorated method that must pass to declared the stress invocation of that method to be declared passed. Range: 0.0-1.0. Default value is provided by environment variable: StressPassThreshold and if undefined then by {@link DefaultStressOptions}.
+ */
+export interface StressOptions {
+	runtime?: number;
+	dop?: number;
+	iterations?: number;
+	passThreshold?: number;
+}
+
+/**
+ * The default values for StressOptions.
+ */
+export const DefaultStressOptions: StressOptions = { runtime: 120, dop: 4, iterations: 10, passThreshold: 0.95 };
+
+/**
+ * Defines the shape of stress result object
+ */
+export interface StressResult {
+	numPasses: number;
+	fails: Error[];
+	errors: Error[];
+}
+
+
+/**
+* This simulates a sleep where the thread is suspended without spinning for a given number of milliseconds before resuming
+*/
+export async function sleep(ms: number) {
+	return await (async () => {
+		return await new Promise((undefined) => setTimeout(undefined, ms));
+	})();
+}
+
+/**
+* This is just a synonym for sleep(0). This has the affect of yielding to other operations.
+*/
+export async function bear() {
+	return await sleep(0);
+}
+
+/**
+ * A class with methods that help to implement the stressify decorator.
+ * Keeping the core logic of stressification in one place as well as allowing this code to use
+ * other decorators if needed.
+ */
+class Stress {
+	// number of iterations.
+	@IsInt()
+	@Min(0)
+	@Max(1000000)
+	iterations?: number = DefaultStressOptions.iterations;
+
+	// seconds
+	@IsInt()
+	@Min(0)
+	@Max(72000)
+	runtime?: number = DefaultStressOptions.runtime;
+
+	// degree of parallelism
+	@IsInt()
+	@Min(1)
+	@Max(20)
+	dop?: number = DefaultStressOptions.dop;
+
+	// threshold for fractional number of individual test passes fo total executed to declare the stress test passed. This is a fraction between 0 and 1.
+	@Min(0)
+	@Max(1)
+	passThreshold?: number = DefaultStressOptions.passThreshold;
+
+	/**
+	 * Constructor allows for construction with a bunch of optional parameters
+	 *
+	 * @param runtime - see {@link StressOptions}.
+	 * @param dop - see {@link StressOptions}.
+	 * @param iterations - see {@link StressOptions}.
+	 * @param passThreshold - see {@link StressOptions}.
+	 */
+	constructor({ runtime = parseInt(process.env.StressRuntime), dop = parseInt(process.env.StressDop), iterations = parseInt(process.env.StressIterations), passThreshold = parseFloat(process.env.StressPassThreshold) }: StressOptions = DefaultStressOptions) {
+		trace(`parameters to Stress constructor: runtime=${runtime}, dop=${dop}, iterations=${iterations}, passThreshold=${passThreshold}`);
+		trace(`default properties of this Stress object: this.runtime=${this.runtime}, this.dop=${this.dop}, this.iterations=${this.iterations}, this.passThreshold=${this.passThreshold}`);
+		let x: number;
+		x = this.nullCoalesce(runtime, this.runtime); this.runtime = x;
```

Comment:
nit: Not a fan of multiple statements on a single line. Makes it hard to see what's actually being done

Reply:
removed.

Reply:
fixed.

---
In reply to: [285192702](https://github.com/microsoft/azuredatastudio/pull/5499#discussion_r285192702) [](ancestors = 285192702)